pkg/repository.go

```diff
@@ -0,0 +1,76 @@
+package pkg
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+	log "github.com/sirupsen/logrus"
+)
+
+//struct which represent mainly interaction with redis
+type RedisRepository struct {
+	Client *redis.Client
+	Ctx  context.Context
+}
+
+
+//should update existing history with order or create a new one 
+func (r *RedisRepository) SaveOrder(userId string, order Order) error {
+	val, err := r.Client.Get(r.Ctx, (userId)).Result()
+	if err != nil {
+		//create new user with user history 
+		history := OrderHistory{Orders: []Order{order}}
+		r.saveHistory(userId,&history)
+	}else {
+		// update existing user history 
+		var history OrderHistory
+		json.Unmarshal([]byte(val), &history)
+		history.add(order)
+		r.saveHistory(userId, &history)	
```

Comment:
don't swallow error from `saveHistory`.
use `if err:= r.saveHistory(userId, &history); err != nil { return err }`
So every error is not lost